### ISSUE TITLE
CKV2_AWS_44 false positive against terraform_plan framework

### ISSUE THREAD

**Issue by citrix-eg** [bridgecrewio/checkov#4078]
**Describe the issue**
When using a common external module (https://registry.terraform.io/modules/terraform-aws-modules/vpc/aws), a checkov scan against the generated plan results in a false positive against the aforementioned check for any subnets created by the module:

```
module "vpc" {
  source  = "terraform-aws-modules/vpc/aws"
  version = "3.14.2"

  name = var.environment
  cidr = "10.0.0.0/19"

  azs = [
    "us-east-2a",
    "us-east-2b",
    "us-east-2c"
  ]

  manage_default_security_group = false #checkov:skip=CKV2_AWS_12:This is zeroed out elsewhere
  private_subnets               = ["10.0.0.0/22", "10.0.4.0/22", "10.0.8.0/22"]
  private_subnet_tags = {
    boundary = "private"
  }

  enable_nat_gateway     = true
  one_nat_gateway_per_az = true #checkov:skip=CKV2_AWS_19:The EIP is attached to a NGW by the module

  public_subnets = ["10.0.26.0/23", "10.0.28.0/23", "10.0.30.0/23"] # 510 hosts
  public_subnet_tags = {
    boundary = "public"
  }
  create_igw              = true
  map_public_ip_on_launch = false

  enable_flow_log                                 = true #checkov:skip=CKV2_AWS_11:We are enabling it here. Likely false positive
  create_flow_log_cloudwatch_iam_role             = true #checkov:skip=CKV_AWS_111:There is no way to configure the resource restriction via this module
  create_flow_log_cloudwatch_log_group            = true #checkov:skip=CKV_AWS_158:We are passing an encryption key below. Likely false positive
  flow_log_cloudwatch_log_group_kms_key_id        = aws_kms_key.cloudwatch_logs.arn
  flow_log_max_aggregation_interval               = 60
  flow_log_cloudwatch_log_group_retention_in_days = 7
  tags = {
    component = "foo"
  }
}
```

**Version (please complete the following information):**
 - 2.2.157 on macos and debian buster

**Additional context**
This only happens with plan checks.

**Comment by gruebel** [comment]
hey @citrix-eg thanks for reaching out.
I tested it and all route tables pass
```shell
Check: CKV2_AWS_44: "Ensure AWS route table with VPC peering does not contain routes overly permissive to all traffic"
	PASSED for resource: module.vpc.aws_route.private_nat_gateway[0]
	File: /tf_vpc.json:196-213
```
